scripts/generate_dump

```diff
@@ -195,8 +195,9 @@ save_cmd() {
         if $NOOP; then
             echo "${timeout_cmd} $cmd $redirect '$filepath'"
         else
-            eval "${timeout_cmd} $cmd" "$redirect" "$filepath"
-            if [ $? -ne 0 ]; then
```

Comment:
This code looks good to me. Wondering where does it fail.

Reply:
oh, I see, it can fail when there's timeout.

Reply:
Didn't look deep into it. One possibility is that command timed out?

Reply:
If you were asking what the issue with this old structure. if line 198 returns non-zero code, then it would trigger the error handling and bypass the handling at 199.

Reply:
got it!